sklearn/model_selection/_validation.py

```diff
@@ -758,27 +758,50 @@ def _score(estimator, X_test, y_test, scorer, error_score="raise"):
     """
     if isinstance(scorer, dict):
         # will cache method calls if needed. scorer() returns a dict
-        scorer = _MultimetricScorer(**scorer)
+        scorer = _MultimetricScorer(scorers=scorer, raise_exc=(error_score == "raise"))
 
     try:
         if y_test is None:
             scores = scorer(estimator, X_test)
         else:
             scores = scorer(estimator, X_test, y_test)
     except Exception:
-        if error_score == "raise":
+        if isinstance(scorer, _MultimetricScorer):
+            assert error_score == "raise", (
+                "If `_MultimetricScorer` raises exception, the `error_score`"
+                " parameter should be equal to 'raise'."
+            )
```

Comment:
I think we do not need to check this because we will are going to raise anyways:

```python
except Exception:
    if error_score == "raise":
        raise
    elif not isinstance(scorer, _MultimetricScorer):
        ...
```